src/test/java/mate/academy/onlinebookstore/service/ShoppingCartServiceTest.java

```diff
@@ -0,0 +1,329 @@
+    }
+
+    @Test
+    @DisplayName("Verify save() method add cart items works when given an existing cart item")
+    public void addCartItem_GivenCartItemWithExistingBook_ShouldIncreaseQuantity() {
+        User user = getUserTest1();
+        userRepository.save(user);
+
+        ShoppingCart shoppingCart = new ShoppingCart()
+                .setId(1L)
+                .setUser(user);
+        shoppingCartRepository.save(shoppingCart);
+
+        Book book = getRedBookWithPrice19_99();
+        bookRepository.save(book);
+
+        CartItem cartItem = new CartItem()
+                .setId(1L)
+                .setShoppingCart(shoppingCart)
+                .setBook(book)
+                .setQuantity(RED_BOOK_QUANTITY);
+        cartItemService.save(cartItem);
+
+        shoppingCart.setCartItems(Collections.singleton(cartItem));
+
+        AddBookToShoppingCartDto addBookToShoppingCartDto = new AddBookToShoppingCartDto();
+        addBookToShoppingCartDto.setBookId(book.getId());
+        addBookToShoppingCartDto.setTitle(book.getTitle());
+        addBookToShoppingCartDto.setAuthor(book.getAuthor());
+        addBookToShoppingCartDto.setQuantity(RED_BOOK_QUANTITY);
+
+        CartItemDto cartItemDto = new CartItemDto();
+        cartItemDto.setId(cartItem.getId());
+        cartItemDto.setBookId(cartItem.getBook().getId());
+        cartItemDto.setBookTitle(cartItem.getBook().getTitle());
+        cartItemDto.setQuantity(cartItem.getQuantity() + 1);
+
+        ShoppingCartDto shoppingCartDto = new ShoppingCartDto()
+                .setId(shoppingCart.getId())
+                .setUserId(shoppingCart.getUser().getId())
+                .setCartItems(List.of(cartItemDto));
+
+        BookDto bookDto = getRedBookDtoWithPrice19_99();
+
+        Mockito.when(userRepository.findByEmail(user.getEmail())).thenReturn(Optional.of(user));
+        Mockito.when(shoppingCartRepository.findByUserId(user.getId())).thenReturn(shoppingCart);
+        Mockito.when(bookService.findById(addBookToShoppingCartDto.getBookId()))
+                .thenReturn(bookDto);
+        doReturn(List.of(cartItem)).when(cartItemService)
+                .findByShoppingCartId(shoppingCart.getId());
+        Mockito.when(cartItemService.save(cartItem)).thenReturn(cartItemDto);
+        Mockito.when(shoppingCartMapper.toDto(shoppingCart)).thenReturn(shoppingCartDto);
+        ShoppingCartDto shoppingCartDtoActual = shoppingCartService
+                .addCartItem(addBookToShoppingCartDto, user.getEmail());
+        assertNotNull(shoppingCartDtoActual);
+        assertEquals(shoppingCartDtoActual, shoppingCartDto);
+    }
+
+    @Test
+    @DisplayName("Verify save() method add cart items works when given an existing cart item")
+    public void addCartItem_GivenCartItemWithNewBook_ShouldAddNewCartItem() {
+        User user = getUserTest1();
+        userRepository.save(user);
+
+        ShoppingCart shoppingCart = new ShoppingCart()
```

Comment:
same, do something, 70 lines of code for test method makes it harder to read than code itself